src/main/groovy/org/gradle/android/AndroidCacheFixPlugin.groovy

```diff
@@ -162,270 +157,20 @@ class AndroidCacheFixPlugin implements Plugin<Project> {
         workaroundsBuilder.build()
     }
 
-    /**
-     * The following are kept for reference purposes.
-     */
-
-    /**
-     * Fix {@link org.gradle.api.tasks.compile.CompileOptions#getBootClasspath()} introducing relocatability problems for {@link AndroidJavaCompile}.
-
-    @AndroidIssue(introducedIn = "3.0.0", fixedIn = ["3.1.0-alpha06", "3.1.1", "3.1.2", "3.1.3", "3.1.4", "3.2.0-alpha01"], link = "https://issuetracker.google.com/issues/68392933")
-    static class AndroidJavaCompile_BootClasspath_Workaround implements Workaround {
-        @CompileStatic(TypeCheckingMode.SKIP)
-        @Override
-        void apply(WorkaroundContext context) {
-            def project = context.project
-            project.tasks.withType(AndroidJavaCompile) { AndroidJavaCompile task ->
-                task.inputs.property "options.bootClasspath", ""
-                // Override workaround introduced in 3.1.0-alpha02
-                task.inputs.property "options.bootClasspath.filtered", ""
-                task.inputs.files({
-                        DeprecationLogger.whileDisabled({
-                            //noinspection GrDeprecatedAPIUsage
-                            task.options.bootClasspath?.split(File.pathSeparator)
-                        } as Factory)
-                    })
-                    .withPathSensitivity(PathSensitivity.RELATIVE)
-                    .withPropertyName("options.bootClasspath.workaround")
-                    .optional(true)
-            }
-        }
-    }
-     */
+    private enum Warnings {
+        MAYBE_SUPPORTED_ANDROID_VERSION("WARNING: Android plugin ${CURRENT_ANDROID_VERSION} has not been tested with this version of the Android cache fix plugin, although it may work.  We test against only the latest patch release versions of Android Gradle plugin: ${SUPPORTED_ANDROID_VERSIONS.join(", ")}.  If ${CURRENT_ANDROID_VERSION} is newly released, we may not have had a chance to release a version tested against it yet.  Proceed with caution.  You can suppress this warning with with -D${IGNORE_VERSION_CHECK_PROPERTY}=true.")
 
-    /**
-     * Filter the Java annotation processor output folder from compiler arguments to avoid absolute path.
+        private final String warning
+        private final AtomicBoolean warned = new AtomicBoolean()
```

Comment:
This is cool! For some reason, I thought that the Plugin object would be created once per project where it is applied.